akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -167,6 +175,7 @@ object GraphStageLogic {
       behaviour = receive
     }
 
+    private[this] var watching = Set.empty[ActorRef]
```

Comment:
The default set is rather minimal for up to four entries and I switched to the cached one.